Leetcode/framework/template.hpp

```diff
@@ -70,80 +71,128 @@ class Planner1Result : public Result {
         std::vector<std::vector<int>> values2;
     };
     
-
-    explicit Planner1Result(std::shared_ptr<Variable> variable)
+    explicit Planner1Result(std::shared_ptr<const Variable> variable)
         : _variable(variable) {}
     ~Planner1Result() = default;
     
     void UpdateResult();
     void ClearResult();
     void Variable2Result();
-    std::pair<std::shared_ptr<Variable>, std::shared_ptr<RelativeData>>
-        GetResult() const { return _result; }
+    auto GetResult() const { return _result; }
```

Comment:
https://google.github.io/styleguide/cppguide.html#Type_deduction
"Use type deduction only if it makes the code clearer to readers who aren't familiar with the project, or if it makes the code safer. Do not use it merely to avoid the inconvenience of writing an explicit type."
感觉这就是偷懒的用法。函数定义用 auto 调用的时候会引起迷惑。你可以在 class 里定义一个 type。